File: frontend/app/services/groups.py
from typing import List, Dict, Optional
from .base_service import BaseService

class GroupService(BaseService):
    def __init__(self):
        super().__init__('/api/groups')
    
    def get_all(self) -> List[Dict]:
        """Get all groups with their accounts"""
        try:
            result = self._handle_request('get', self.endpoint)
            return result if result else []
        except Exception as e:
            print(f"Error getting groups: {str(e)}")
            return []
    
    def create(self, data: Dict) -> Optional[Dict]:
        """Create a new group"""
        try:
            return self._handle_request('post', self.endpoint, data)
        except Exception as e:
            print(f"Error creating group: {str(e)}")
            return None

    def get_by_id(self, group_id: int) -> Optional[Dict]:
        """Get a specific group by ID"""
        try:
            return self._handle_request('get', f"{self.endpoint}/{group_id}")
        except Exception as e:
            print(f"Error getting group: {str(e)}")
            return None

    def assign_account(self, group_id: int, account_id: int) -> bool:
        """Assign an account to a group"""
        try:
            result = self._handle_request(
                'post', 
                f"{self.endpoint}/{group_id}/accounts/{account_id}"
            )
            return bool(result and result.get('message'))
        except Exception as e:
            print(f"Error assigning account to group: {str(e)}")
            return False

File: frontend/app/static/js/accounts/groups.js
export class AccountGroupManager {
    constructor() {
        this.groupsContainer = document.getElementById('account-groups');
        this.csrfToken = document.querySelector('meta[name="csrf-token"]')?.content;
    }

    init() {
        this.attachEventListeners();
    }

    attachEventListeners() {
        document.getElementById('create-group-btn')?.addEventListener('click', 
            () => this.showCreateGroupModal());

        document.querySelectorAll('.assign-to-group').forEach(button => {
            button.addEventListener('click', (e) => this.handleGroupAssignment(e));
        });

        // Add form submit handler
        const form = document.getElementById('create-group-form');
        if (form) {
            form.addEventListener('submit', async (e) => {
                e.preventDefault();
                await this.createGroup(new FormData(e.target));
            });
        }
    }

    async showCreateGroupModal() {
        const modal = document.getElementById('create-group-modal');
        if (!modal) return;
        new bootstrap.Modal(modal).show();
    }

    async createGroup(formData) {
        try {
            const response = await fetch('/api/groups/', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                    'X-CSRFToken': this.csrfToken
                },
                body: JSON.stringify({
                    name: formData.get('name'),
                    description: formData.get('description')
                })
            });

            const data = await response.json();
            
            if (!response.ok) {
                throw new Error(data.error || 'Failed to create group');
            }

            window.location.reload();
        } catch (error) {
            console.error('Error creating group:', error);
            alert(error.message || 'Failed to create group. Please try again.');
        }
    }

    async handleGroupAssignment(event) {
        const accountId = event.target.dataset.accountId;
        const groupId = event.target.dataset.groupId;

        try {
            const response = await fetch(`/api/groups/${groupId}/accounts/${accountId}`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                    'X-CSRFToken': this.csrfToken
                }
            });

            const data = await response.json();
            
            if (!response.ok) {
                throw new Error(data.error || 'Failed to assign group');
            }

            window.location.reload();
        } catch (error) {
            console.error('Error assigning group:', error);
            alert(error.message || 'Failed to assign group. Please try again.');
        }
    }
}